File: 231.Power-of-Two.py
# https://leetcode.com/problems/basic-calculator-ii/description/
#
# algorithms
# Easy (41.49%)
# Total Accepted:    207,010
# Total Submissions: 498,902

class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n <= 0:
            return False
        return (math.log10(n) / math.log10(2)) % 1 == 0

File: 231.Power-of-Two.java
// https://leetcode.com/problems/power-of-two/
//
// algorithms
// Easy (41.95%)
// Total Accepted: 227,879
// Total Submissions: 543,208

// 这道题有很多种做法，包括取对数相除都是好方法
// 这种做法应该是最好的
// n & (n - 1) 会消去 n 的二进制表示中的最右边的 1
// 如果是 2 的倍数的话，则说明二进制中只有一个 1
class Solution {
    public boolean isPowerOfTwo(int n) {
        return n > 0 && ((n & (n - 1)) == 0);
    }
}